lib/apiservers/service/restapi/handlers/vch_get.go

```diff
@@ -0,0 +1,412 @@
+// Copyright 2017 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package handlers
+
+import (
+	"bytes"
+	"context"
+	"encoding/pem"
+	"fmt"
+	"net"
+	"net/url"
+	"strings"
+
+	"github.com/docker/docker/opts"
+	"github.com/go-openapi/runtime/middleware"
+	"github.com/go-openapi/strfmt"
+	"github.com/vmware/govmomi/vim25/types"
+
+	"github.com/vmware/vic/lib/apiservers/service/models"
+	"github.com/vmware/vic/lib/apiservers/service/restapi/handlers/util"
+	"github.com/vmware/vic/lib/apiservers/service/restapi/operations"
+	"github.com/vmware/vic/lib/config"
+	"github.com/vmware/vic/lib/config/executor"
+	"github.com/vmware/vic/lib/install/data"
+	"github.com/vmware/vic/lib/install/management"
+	"github.com/vmware/vic/lib/install/validate"
+	"github.com/vmware/vic/pkg/version"
+	"github.com/vmware/vic/pkg/vsphere/vm"
+)
+
+// VCHGet is the handler for inspecting a VCH
+type VCHGet struct {
+}
+
+// VCHGet is the handler for inspecting a VCH within a Datacenter
+type VCHDatacenterGet struct {
+}
+
+func (h *VCHGet) Handle(params operations.GetTargetTargetVchVchIDParams, principal interface{}) middleware.Responder {
+	d := buildData(
+		url.URL{Host: params.Target},
+		principal.(Credentials).user,
```

Comment:
is wondering how the principal is set. is that the credential to login vic-machine service?

Reply:
Currently, we've only implemented basic authentication (where a username and password are passed via an `Authorization` header using the `Basic` scheme: https://tools.ietf.org/html/rfc7617)

Relevant code:
* https://github.com/vmware/vic/blob/feature/vic-machine-service/lib/apiservers/service/restapi/configure_vic_machine.go#L55
* https://github.com/vmware/vic/blob/feature/vic-machine-service/lib/apiservers/service/restapi/handlers/authentication.go

Implementing an additional type of authentication is tracked by #6033